Replace native javascript with angularJS code

jduehring opened the pull request:
This PR refers to #2786 and replaces some javascript code which should not have made it into the code. It affects the admin-ui-frontend code responsible for displaying the embed code for each episode. 

This fixes #2786

### Your pull request should…

* [x] have a concise title
* [x] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [x] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [x] include migration scripts and documentation, if appropriate
* [x] pass automated tests
* [x] have a clean commit history
* [x] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

jduehring committed:
basic idea for embedding directive

jduehring committed:
rewrite plain js to angularjs

jduehring committed:
merge develop

jduehring committed:
fix merge

Commit:
Merge branch 'develop' into embedCodeAngular

Commit:
remove comment

Commit:
Merge branch 'develop' into embedCodeAngular

Commit:
Merge branch 'develop' into embedCodeAngular

jduehring committed:
delete unnecessary test file

jduehring committed:
remove unnecessary test file

lkiesow commented on modules/admin-ui-frontend/app/scripts/modules/events/controllers/eventsController.js:
```diff
@@ -159,6 +159,46 @@ angular.module('adminNg.controllers')
+    $scope.embedCode = '';
+
+    $scope.getEmbedCode = function (size, id) {
+      $scope.selectedBox = size;
+      var engageUrl = '<SERVER_URL>';
+      var embedID = id;
+      $http.get('/api/info/organization/properties/engageuiurl')
+      .then(function(response) {
+        engageUrl = response.data['org.opencastproject.engage.ui.url'];
+        var url = engageUrl + '/play/' + embedID;
+        var sizeArray = size.split('x');
+        $scope.embedCode = '<iframe allowfullscreen src="' + url +
+        '" style="border:0px #FFFFFF none;" name="Player" scrolling="no"' +
+         ' frameborder="0" marginheight="0px" marginwidth="0px" width="' +
+        sizeArray[0] + '" height="' + sizeArray[1] + '"></iframe>';
+      });
```
Instead of getting the data from the external API, I suggest a slight change to this since the admin interface already has a resource for this. The patch would be:

```diff
diff --git a/modules/admin-ui-frontend/app/scripts/modules/events/controllers/eventsController.js b/modules/admin-ui-frontend/app/scripts/modules/events/controllers/eventsController.js
index cf8a3ec117..e1562f46fb 100644
--- a/modules/admin-ui-frontend/app/scripts/modules/events/controllers/eventsController.js
+++ b/modules/admin-ui-frontend/app/scripts/modules/events/controllers/eventsController.js
@@ -24,9 +24,9 @@
 angular.module('adminNg.controllers')
 .controller('EventsCtrl', ['$scope', 'Stats', 'Table', 'EventsResource', 'ResourcesFilterResource',
   'ResourcesListResource', 'Notifications', 'ConfirmationModal', 'RelativeDatesService', 'AuthService',
-  'CommentResource','$http',
+  'CommentResource','IdentityResource',
   function ($scope, Stats, Table, EventsResource, ResourcesFilterResource, ResourcesListResource, Notifications,
-    ConfirmationModal, RelativeDatesService, AuthService, CommentResource, $http) {
+    ConfirmationModal, RelativeDatesService, AuthService, CommentResource, IdentityResource) {
 
     $scope.stats = Stats;
 
@@ -162,18 +162,24 @@ angular.module('adminNg.controllers')
     $scope.embedCode = '';
 
     $scope.getEmbedCode = function (size, id) {
-      $scope.selectedBox = size;
-      var engageUrl = '<SERVER_URL>';
-      var embedID = id;
-      $http.get('/api/info/organization/properties/engageuiurl')
-      .then(function(response) {
-        engageUrl = response.data['org.opencastproject.engage.ui.url'];
-        var url = engageUrl + '/play/' + embedID;
-        var sizeArray = size.split('x');
+      const identity = IdentityResource.get();
+      identity.$promise.then(function (info) {
+
+        // check if the engage URL is configured
+        let engageUrl = window.location.origin;
+        if (info && info.org && info.org.properties && info.org.properties) {
+          engageUrl = info.org.properties['org.opencastproject.engage.ui.url'] || engageUrl;
+        }
+
+        const url = engageUrl + '/play/' + id;
+        const sizeArray = size.split('x'),
+              width = sizeArray[0],
+              height = sizeArray[1];
+        $scope.selectedBox = size;
         $scope.embedCode = '<iframe allowfullscreen src="' + url +
-        '" style="border:0px #FFFFFF none;" name="Player" scrolling="no"' +
-         ' frameborder="0" marginheight="0px" marginwidth="0px" width="' +
-        sizeArray[0] + '" height="' + sizeArray[1] + '"></iframe>';
+          '" style="border: none;" name="Player" scrolling="no"' +
+          ' frameborder="0" marginheight="0px" marginwidth="0px" width="' +
+          width + '" height="' + height + '"></iframe>';
       });
     };
 
```

I could apply this patch to this the pull request if you like me to.

jduehring replied:
Looks good! Feel free to apply the patch.

lkiesow commented on modules/admin-ui-frontend/app/scripts/shared/partials/modals/embedding-code.html:
```diff
@@ -1,85 +1,41 @@
-  <span id="id_video" class="embedded-code-no-visible">{{id}}</span>
-  <div id='' class="embedded-code-video"><textarea id='social_embed-textarea' class="social_embed-textarea embedded-code-textarea" rows='4' cols='1'></textarea></div>
+
+  <div id='' class="embedded-code-video">
```
```suggestion
  <div class="embedded-code-video">
```

lkiesow commented on modules/admin-ui-frontend/bower.json:
```diff
@@ -31,5 +31,8 @@
   "resolutions": {
     "angular": "1.8.2"
-  }
+  },
+"resolutions": {
+  "angular-translate": "2.18.3"
+}
 }
```
I suspect this is an unwanted change, is it?

jduehring replied:
True, I'll remove it.

lkiesow reviewed (changes requested):
Works. A few suggestions for improvements.

lkiesow committed:
Get Configuration from IdentityResource

Instead of getting the data from the external API, this patch uses the
already existing method of getting the Engage UI URL configuration via
the IdentityResource.

jduehring committed:
Update modules/admin-ui-frontend/app/scripts/shared/partials/modals/embedding-code.html

remove empty id field

Co-authored-by: Lars Kiesow <lkiesow@uos.de>

jduehring committed:
remove unwanted dependency

gregorydlogan commented:
Is this supposed to end up in OC 11?  If so please change the merge target!

lkiesow reviewed (approved):
> Looks good! Feel free to apply the patch.

Patch applied. This should be ready to go once it went through the technical meeting next Tuesday.
